plugins/modules/aci_access_span_src_group.py

```diff
@@ -0,0 +1,326 @@
+method:
+  description: The HTTP method used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: POST
+response:
+  description: The HTTP response from the APIC
+  returned: failure or debug
+  type: str
+  sample: OK (30 bytes)
+status:
+  description: The HTTP status from the APIC
+  returned: failure or debug
+  type: int
+  sample: 200
+url:
+  description: The HTTP url used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: https://10.11.12.13/api/mo/uni/tn-production.json
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.aci.plugins.module_utils.aci import ACIModule, aci_argument_spec, aci_annotation_spec, aci_owner_spec
+
+
+def main():
+    argument_spec = aci_argument_spec()
+    argument_spec.update(aci_annotation_spec())
+    argument_spec.update(aci_owner_spec())
+    argument_spec.update(
+        source_group=dict(type="str", aliases=["name", "src_group"]),  # Not required for querying all objects
+        description=dict(type="str", aliases=["descr"]),
+        admin_state=dict(type="bool"),
+        filter_group=dict(type="str"),
+        destination_group=dict(type="str", aliases=["dst_group"]),
+        state=dict(type="str", default="present", choices=["absent", "present", "query"]),
+        name_alias=dict(type="str"),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "absent", ["source_group"]],
+            ["state", "present", ["source_group", "destination_group"]],
+        ],
+    )
+
+    aci = ACIModule(module)
+
+    source_group = module.params.get("source_group")
+    description = module.params.get("description")
+    admin_state = aci.boolean(module.params.get("admin_state"), "enabled", "disabled")
+    filter_group = module.params.get("filter_group")
+    destination_group = module.params.get("destination_group")
+    state = module.params.get("state")
+    name_alias = module.params.get("name_alias")
+
+    aci.construct_url(
+        root_class=dict(
+            aci_class="infra",
+            aci_rn="infra",
+        ),
+        subclass_1=dict(
+            aci_class="spanSrcGrp",
+            aci_rn="srcgrp-{0}".format(source_group),
+            module_object=source_group,
+            target_filter={"name": source_group},
+        ),
+        child_classes=["spanSpanLbl", "spanRsSrcGrpToFilterGrp"],
+    )
+
+    aci.get_existing()
+
+    if state == "present":
+        # Create new child configs payload
+        filter_group_tdn = "uni/infra/filtergrp-{0}".format(filter_group)
+        child_configs = [{"spanSpanLbl": {"attributes": {"name": destination_group}}}]
+        if filter_group:
+            child_configs.append({"spanRsSrcGrpToFilterGrp": {"attributes": {"tDn": filter_group_tdn}}})
+
+        # Validate if existing and remove child objects when do not match provided configuration
```

Comment:
For now yes, but they could be used for any api call / request to bypass the coupled behaviour attached to executing a request.

Reply:
This is a possible other option, the downside I see to "failing and displaying a message to delete the child config and add a new one" is that it would require the following:

- create additional 5 modules for all the child object manipulation ( spanRsSrcGrpToFilterGrp on group level, spanRsSrcToFilterGrp on source level, spanRsSrcToEpg on source level, spanRsSrcToL3extOut on source level ). Then from consistency point of view you would probably also want to add spanSpanLbl on group level. 
- more actions needed for user to make a change 
- increase in request amount

 Perhaps something we should discuss in more depth?